packages/webapp/pages/earnings.tsx

```diff
@@ -165,8 +169,17 @@ const Earnings = (): ReactElement => {
     transactionsQuery;
 
   const hasTransactions = (transactions?.pages?.[0]?.edges?.length || 0) > 0;
+  const isPageReady = router?.isReady && isAuthReady;
 
-  if (!user) {
+  useEffect(() => {
+    if ((hasAccessToCores(user) && isLoggedIn) || !isPageReady) {
+      return;
+    }
```

Comment:
Can you split into multiple early returns, it will be much more readable?

Reply:
just to one for each condition, much easier to follow

Reply:
actually I think you don't need `isLoggedIn` since `hasAccessToCores` should return false on anonymous user?

Reply:
For anon, `coresRole` is undefined. Maybe update the `verifyCoresRole` to return false if it is null or undefined

Reply:
Refactored and removed `isLoggedIn`